internal/testing/types/string.go

```diff
@@ -57,17 +57,21 @@ func (t StringType) ValueFromTerraform(ctx context.Context, in tftypes.Value) (a
 		return nil, err
 	}
 	return String{
-		String:    types.String{Value: s},
-		CreatedBy: t,
+		InternalString: types.String{Value: s},
+		CreatedBy:      t,
 	}, nil
 }
 
 type String struct {
-	types.String
+	InternalString types.String
```

Comment:
I had to change the name we use for the embedded struct here, as now it was clashing with the method name.